### ISSUE TITLE
IonicModule.forRoot() is missing deep-link-config that is refered to via a variable

### ISSUE THREAD

**Issue by fs5** [ionic-team/ionic-app-scripts#1264]
#### Short description of the problem:

Transpilation of the following snippet is incorrect (pseudo-code from app.module.ts)

```typescript
const DEEP_LINK_CONFIG: DeepLinkConfig = {
  links: [
    { component: TabsContainer, name: 'TabsContainer', segment: 'tabs-container' },
    { component: EventListPage, name: 'EventList', segment: 'event-list' }
  ]
};

const IONIC_CONFIG: any = { locationStrategy: 'hash' };

@NgModule({
  imports: [
    ...,
    IonicModule.forRoot(AppComponent, IONIC_CONFIG, DEEP_LINK_CONFIG)
  ],
  declarations: ...
})
export class AppModule { }
```

`ionic cordova build browser` produces

```javascript
var DEEP_LINK_CONFIG = {
    links: [
        { component: __WEBPACK_IMPORTED_MODULE_6__containers__["a" /* TabsContainer */], name: 'TabsContainer', segment: 'tabs-container' },
        { component: __WEBPACK_IMPORTED_MODULE_7__pages__["a" /* EventListPage */], name: 'EventList', segment: 'event-list' }
    ]
};
var IONIC_CONFIG = { locationStrategy: 'hash' };
var AppModule = /** @class */ (function () {
    function AppModule() {
    }
    AppModule = __decorate([
        Object(__WEBPACK_IMPORTED_MODULE_0__angular_core__["L" /* NgModule */])({
            imports: [
...
                __WEBPACK_IMPORTED_MODULE_4_ionic_angular__["c" /* IonicModule */].forRoot(
__WEBPACK_IMPORTED_MODULE_5__app_component__["a" /* AppComponent */], 
IONIC_CONFIG, 
/* this is wrong */ { links: [] })
            ],
            declarations: ...
        })
    ], AppModule);
    return AppModule;
}());
```

#### What behavior are you expecting?
```javascript
__WEBPACK_IMPORTED_MODULE_4_ionic_angular__["c" /* IonicModule */].forRoot(
__WEBPACK_IMPORTED_MODULE_5__app_component__["a" /* AppComponent */], 
IONIC_CONFIG, 
/* this would be correct */ DEEP_LINK_CONFIG)
```
The last parameter should be `DEEP_LINK_CONFIG` but is erroneously defaulted to `{
                    links: []
                }`

**Steps to reproduce:**
see above

**Which @ionic/app-scripts version are you using?**

3.0.0

The issue does not occur with 2.1.4

**Comment by danbucholtz** [comment]
This is likely a bug. I'll investigate. Any reason you haven't migrated to using the `@IonicPage()` decorator for deep linking?

Thanks,
Dan

**Comment by RichardM99** [comment]
I am having the same issue. Refering to the links via a variable not showing the segments/paths, however  refering to them directly does. e.g.

```
// not working
const deepLinkConfig: DeepLinkConfig = {
  links: [
    { component: SomeComponent, name: 'SomeComponent', segment: 'some-component'}
  ]
};

imports: [
    ChartsModule,
    BrowserModule,
    HttpModule,
    IonicModule.forRoot(
      MyApp,
      {
        iconMode: 'md',
        mode    : 'md',
        useHash : false,
        tabsPlacement: 'top'
      },
      deepLinkConfig,
    ),
    IonicStorageModule.forRoot()
  ],

//working 
 imports: [
    ChartsModule,
    BrowserModule,
    HttpModule,
    IonicModule.forRoot(
      MyApp,
      {
        iconMode: 'md',
        mode    : 'md',
        useHash : false,
        tabsPlacement: 'top'
      },
      {
       links: [
           { component: SomeComponent, name: 'SomeComponent', segment: 'some-component'},
        ]
      },
    ),
    IonicStorageModule.forRoot()
  ],
```
Using version 3.0.0

**Comment by danbucholtz** [comment]
This will be fixed in 3.0.1 releasing on Wednesday.

Thanks,
Dan

**Comment by fs5** [comment]
Thank you for the fix. It works for the case as described above. But it still fails when the deep-link-config is imported from another typescript file:

```javascript
import { DEEP_LINK_CONFIG } from "./app.routes";

const IONIC_CONFIG: any = { locationStrategy: 'hash' };

@NgModule({
  imports: [
    ...,
    IonicModule.forRoot(AppComponent, IONIC_CONFIG, DEEP_LINK_CONFIG)
  ],
  declarations: ...
})
export class AppModule { }
```
